lib/rust/parser/src/macros/resolver.rs

```diff
@@ -123,18 +123,17 @@ impl<'a> SegmentMap<'a> {
 /// to learn more about the macro resolution steps.
 #[derive(Debug)]
 pub struct Resolver<'s> {
-    blocks:        Vec<Block>,
+    blocks:     Vec<Block>,
     /// The lines of all currently-open blocks. This is partitioned by `blocks`.
-    lines:         Vec<syntax::item::Line<'s>>,
+    lines:      Vec<syntax::item::Line<'s>>,
     /// All currently-open macros. These are partitioned into scopes by `blocks`.
-    macros:        Vec<PartiallyMatchedMacro<'s>>,
+    macros:     Vec<PartiallyMatchedMacro<'s>>,
     /// Segments of all currently-open macros. These are partitioned by `macros`.
-    segments:      Vec<MatchedSegment<'s>>,
+    segments:   Vec<MatchedSegment<'s>>,
     /// Items of all segments of all currently-open macros. These are partitioned by `segments`.
-    items:         Vec<syntax::Item<'s>>,
-    context:       Context,
-    precedence:    syntax::operator::Precedence<'s>,
-    start_segment: bool,
```

Comment:
Looks like the `start_segment` idea wasn't the right way to solve the problem. 287e307083413c81f2c40084a24ad532dff3fa0e (with its `Block` and `NotBlock` patterns) is.